apps/src/templates/sectionProgressV2/ProgressTableV2.jsx

```diff
@@ -3,39 +3,73 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import {studentShape} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
 import StudentColumn from './StudentColumn';
-import ProgressTableHeader from './ProgressTableHeader';
-import ProgressDataV2 from './ProgressDataV2';
 import styles from './progress-table-v2.module.scss';
 import stringKeyComparator from '@cdo/apps/util/stringKeyComparator';
 import {getCurrentUnitData} from '../sectionProgress/sectionProgressRedux';
 import {unitDataPropType} from '../sectionProgress/sectionProgressConstants';
+import ExpandedProgressDataColumn from './ExpandedProgressDataColumn';
+import LessonProgressDataColumn from './LessonProgressDataColumn';
 
 export function ProgressTableV2({
   isSortedByFamilyName,
   sectionId,
   students,
   unitData,
+  expandedLessonIds,
+  setExpandedLessons,
 }) {
   const sortedStudents = React.useMemo(() => {
     return isSortedByFamilyName
       ? students.sort(stringKeyComparator(['familyName', 'name']))
       : students.sort(stringKeyComparator(['name', 'familyName']));
   }, [students, isSortedByFamilyName]);
 
+  const renderedColumns = React.useMemo(() => {
+    const lessons = unitData?.lessons;
+    if (lessons === undefined || lessons.length === 0) {
+      return [];
+    }
+
+    let columns = [];
```

Comment:
This array is no longer populated. The `key={columns.length}` below therefore probably doesn't work as you would expect, but the components don't seem to be doing anything with that at the moment anyway. If this ID becomes necessary you could get it by putting a second parameter in the function passed to `map`.

Reply:
Key is necessary for any list of elements in a div, otherwise you might get weird ordering and behavior on render. I added the correct index.

Reply:
Ah yes, of course, I was just looking for explicit parameters and didn't see a `key` in there.